R/zchunk_L141.hfc_R_S_T_Y.R

```diff
@@ -49,69 +46,206 @@ module_emissions_L141.hfc_R_S_T_Y_DISABLED <- function(command, ...) {
+    # Calculate share of res/com cooling emissions from L144.in_EJ_R_bld_serv_F_Yh
+    # Apply share to HFC emissions
+
+    # Select residential and cooling emisssions from L144.in_EJ_R_bld_serv_F_Yh
+    L141.R_cooling_T_Yh.long <- L144.in_EJ_R_bld_serv_F_Yh %>%
+      filter(service %in% c("comm cooling", "resid cooling") & fuel == "electricity")
+    # Group by GCAM region and ID and year in new data frame (use to calculate share of total later)
+    L141.R_cooling_Yh <-  L141.R_cooling_T_Yh.long %>%  group_by(GCAM_region_ID,year) %>%
+      summarize(total = sum(value))
+    # join in the totals we just calculated and calulate the shares
+    L141.R_cooling_T_Yh.long <- left_join_error_no_match(L141.R_cooling_T_Yh.long,L141.R_cooling_Yh, by = c("GCAM_region_ID", "year")) %>%
+      mutate(share = value / total ) %>%
+      mutate(year = as.numeric(gsub('X','',year))) %>%
+      select(GCAM_region_ID,year,service,share,value)
+
+    # Add res/com cooling share to HFC and make adjustment
+    L141.hfc_R_S_T_Yh_share<- left_join(L141.hfc_R_S_T_Yh.long,
+                                        L141.R_cooling_T_Yh.long ,
+                                        by = c("GCAM_region_ID", "year", "supplysector" = "service")) %>% # there should be NAs.
+                                        #there are sectors that don't have emissions
+      mutate_at(vars(),funs(replace(.,is.na(.),1))) %>% # replace those shares with "1"
+      mutate(emissions = emissions*share)
+
+    # add EDGAR 2008 year, as 2010 to match with Guus
+    TEMP <- subset( L141.hfc_R_S_T_Yh_share, year == 2008 )
+    TEMP$year <- 2010
+    L141.hfc_R_S_T_Yh_share <- bind_rows( L141.hfc_R_S_T_Yh_share, TEMP )
+
+    # process Guus inventory
+    HFC_Inventory_GV <- HFC_Inventory_GV %>% mutate(Species = gsub("-","", Species)) %>%
+      mutate(Species = gsub("4310mee", "43",Species)) %>%
+      dplyr::rename(Guus_tot = Emissions)
+
+    # add Guus values to HFC emissions then calculate scalar
+    L141.hfc_scaler <- L141.hfc_R_S_T_Yh_share %>% group_by(year,Non.CO2) %>%
+      summarize(EDGAR_tot = sum(emissions))%>%
+      arrange(Non.CO2,year) %>%
+      left_join(HFC_Inventory_GV, by = c(year = "Year", Non.CO2 = "Species")) %>% # some entries not in GUUS data, default to scaler =1
+      mutate(scaler = Guus_tot/EDGAR_tot) %>%
+      mutate_at(vars(), funs(replace(., is.na(.), 1))) %>%
+      mutate_at(vars(), funs(replace(., is.infinite(.), 0))) %>%
+      select(year, Non.CO2, scaler)
+
+    # Add scalar to HFC emissions, calculate adjusted emissions
+    L141.hfc_R_S_T_Yh <- L141.hfc_R_S_T_Yh_share %>% left_join(L141.hfc_scaler,
+                                        by = c("year", "Non.CO2") ) %>% # there are NA's, not all sectors have emissions/scalers
+      mutate(adj_emissions = emissions*scaler) %>%
+      group_by(GCAM_region_ID, supplysector, subsector, stub.technology, Non.CO2, year) %>% #this takes SOOOO LONG
```

Comment:
My guess would be failure to ungroup before running the `mutate_at` on line 182.  Ungrouping and using `replace_na` should help.